packages/driver/src/cy/focused.coffee

```diff
@@ -146,7 +147,11 @@ create = (state) ->
     ## is always in focus
     $focused = getFocused()
 
-    if $elements.isFocusable($dom.wrap(el)) && (!$focused || $focused[0] isnt el)
+    if (
+      $elements.isFocusable($dom.wrap(el)) &&
+      (!$focused || $focused[0] isnt el) &&
+      !($visibility.parentHasDisplayNone($dom.wrap(el)) || $dom.wrap(el).css('visibility') is 'hidden')
```

Comment:
hopefully this is the most optimal order of conditionals

Reply:
Can we extract this logic into a function that clearly expresses all the things that these checks do? It's fine if its a looooong name. Otherwise this is extremely dense. I'd accept comments as well but perhaps we can come up with a good name.

Reply:
```suggestion
      !$visibility.parentHasDisplayNone($dom.wrap(el)) &&
      $dom.wrap(el).css('visibility') isnt 'hidden'
```